### ISSUE TITLE
Create virtual screen to use as second screen

### ISSUE THREAD

**Issue by vidister** [any1/wayvnc#7]
Is there a way to automatically create a virtual screen as vnc source?
The background is that I want to use my iPad as a second screen for my computer.

**Comment by any1** [comment]
This is currently not possible, but might be in the future when virtual displays are added to wlroots.

**Comment by rhysperry111** [comment]
The feature for virtual displays has just been merged into Sway. It will be included in the 1.5 release.

See here: https://github.com/swaywm/sway/pull/5216. There currently isn't any documentation (that I could find) but you might be able to derive the options by reading the relevant code here:https://github.com/swaywm/sway/blob/master/sway/commands/create_output.c

**Comment by genofire** [comment]
It works with `sway` \o/ ...
```
swaymsg create_output
swaymsg output "HEADLESS-1" resolution 3840x2160
wayvnc -o HEADLESS-1 0.0.0.0
```
(Importent change the resolution before starting wayvnc ... there are some errors ....)
(alternative for line 2, you could put something equal into `.config/sway/config` or use [kanshi](https://github.com/emersion/kanshi), [wdisplays](https://github.com/cyclopsian/wdisplays))

It is really slow + sometime it closed the connection without reason, and sometimes it crashed.
(is there a option to get logs in wayvnc?)

**Comment by any1** [comment]
There's not a lot of logging, but there is some debug logging available if you run meson with `--buildtype=debug`. A debugger backtrace would help to know why it crashes, in which case it would be better to open a new issue.

It could be "slow" for multiple reasons: network connection, poor client, slow computer or poorly chosen encoding method.

**Comment by ldelossa** [comment]
@vidister I found this because I'm trying to use wayvnc to extend a screen to my Ipad, did this ever work for you?

**Comment by martinetd** [comment]
@ldelossa https://github.com/any1/wayvnc/issues/7#issuecomment-657092231 a few comments above (swaymsg commands to create display and use it with wayvnc) should be what you're trying to do, what's wrong with it?

**Comment by ldelossa** [comment]
More looking for, what app and how did you connect via IPad. Maybe some docs related to getting it to work with a new IPAD.
